src/main/java/com/faforever/client/leaderboard/LeaderboardController.java

```diff
@@ -0,0 +1,369 @@
+        if (leagueEntry == null) {
+          return;
+        }
+        selectLeagueEntry(leagueEntry);
+      })));
+  }
+
+  public void setSeason(LeagueSeasonBean season) {
+    this.season = season;
+
+    seasonLabel.setText(i18n.getOrDefault(season.getNameKey(), String.format("leaderboard.season.%s", season.getNameKey())).toUpperCase());
+    String startDate = season.getStartDate().format(DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM));
+    String endDate = season.getEndDate().format(DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM));
+    seasonDateLabel.setText(i18n.get("leaderboard.seasonDate", startDate, endDate));
+    contentPane.setVisible(false);
+    searchTextField.clear();
+    if (usernamesAutoCompletion != null) {
+      usernamesAutoCompletion.dispose();
+    }
+    leaderboardService.getAllSubdivisions(season.getId()).thenAccept(this::setUsernamesAutoCompletion)
+        .exceptionally(throwable -> {
+          contentPane.setVisible(false);
+          log.warn("Error while loading division list", throwable);
+          notificationService.addImmediateErrorNotification(throwable, "leaderboard.failedToLoad");
+          return null;
+        });
+    setListener(playerService.getCurrentPlayer());
+  }
+
+  private void setUsernamesAutoCompletion(List<SubdivisionBean> subdivisions) {
+    JavaFxUtil.runLater(() -> {
+      majorDivisionPicker.getItems().clear();
+      majorDivisionPicker.getItems().addAll(
+          subdivisions.stream().filter(subdivision -> subdivision.getIndex() == 1).collect(Collectors.toList()));
+    });
+    List<LeagueEntryBean> leagueEntries = new ArrayList<>();
+    List<CompletableFuture<?>> futures = new ArrayList<>();
+    subdivisions.forEach(subdivision -> {
+      CompletableFuture<List<LeagueEntryBean>> future = leaderboardService.getEntries(subdivision);
+      futures.add(future.thenAccept(leagueEntries::addAll));
+    });
+    CompletableFuture.allOf(futures.toArray(new CompletableFuture<?>[0])).thenRun(() -> {
+      usernamesAutoCompletion = TextFields.bindAutoCompletion(searchTextField,
+          leagueEntries.stream().map(leagueEntryBean -> leagueEntryBean.getPlayer().getUsername()).collect(Collectors.toList()));
+      usernamesAutoCompletion.setDelay(0);
+      JavaFxUtil.runLater(() -> contentPane.setVisible(true));
+    });
+  }
+
+  @Override
+  public Tab getRoot() {
+    return leaderboardRoot;
+  }
+
+  private void setListener(PlayerBean player) {
+    playerRatingListener = observable -> updateDisplayedPlayerStats(player);
+    JavaFxUtil.addAndTriggerListener(player.getLeaderboardRatings(), new WeakInvalidationListener(playerRatingListener));
+  }
+
+  @VisibleForTesting
+  protected void updateDisplayedPlayerStats(PlayerBean player) {
+    leaderboardService.getAllSubdivisions(season.getId()).thenAccept(divisions ->
+      leaderboardService.getLeagueEntryForPlayer(player, season.getId()).thenAccept(leagueEntry -> {
```

Comment:
We also use `getLeagueEntryForPlayer` in `processSearchInput` to then select the right tab and select the entry for the player the user searched for

Reply:
Yes there is no need to remove that method just in this case the listener and handing updateDisplayedPlayerStats accepting a player are not needed in the leaderboardController since the value never changes

Reply:
Oh wait, I missunderstood. Yeah, I guess that makes sense